### PR TITLE
Kickoff 5.2.0 development [API-1564]

### PR DESCRIPTION
**Pull request by mdumandag**
This PR contains various fixes to make the client compatible with the 5.2 server.

- Server version used in tests bumped to 5.2.2
- For Compact, field kinds are updated with the up-to-date enum values and they are moved to the serialization.api from compact, to match with the Java side
- Dependency on the enterprise-tests JAR is removed and the self-signed certificates that we depend on moved to the client-side, instead of adding a dependency to the private test artifacts.
- Some reactor related tests that fail on macOS are fixed

### CODE REVIEW

File: hazelcast/serialization/api.py [hunk 1]
```diff
@@ -2593,3 +2594,57 @@ def get_type_name(self) -> str:
...
+    ARRAY_OF_TIME = 22
+    DATE = 23
+    ARRAY_OF_DATE = 24
+    TIMESTAMP = 25
+    ARRAY_OF_TIMESTAMP = 26
+    TIMESTAMP_WITH_TIMEZONE = 27
+    ARRAY_OF_TIMESTAMP_WITH_TIMEZONE = 28
+    COMPACT = 29
+    ARRAY_OF_COMPACT = 30
+    NULLABLE_BOOLEAN = 33
```
Comment (yuce) [inline hazelcast/serialization/api.py:2637]: Adding unsupported fields and 31, 32 may be nice as documentation.

File: tests/unit/reactor_test.py [hunk 2]
```diff
@@ -182,24 +183,29 @@ def test_wake(self):
         self.assertFalse(waker.awake)
         waker.wake()
         self.assertTrue(waker.awake)
+        # Wait until the reader becomes readable
+        select.select([waker._reader], [], [], 10)
```
Comment (yuce) [inline tests/unit/reactor_test.py:187]: Would it make sense to move this to its own function, like: `waker.wait_reader()`?

**Review by yuce — COMMENTED** [review]
Looks good, just two nits.

File: tests/unit/reactor_test.py (hunk 2 above)
Comment (mdumandag) [reply tests/unit/reactor_test.py:187]: I need this just for tests, I don't want to add a function for such a single liner

File: hazelcast/serialization/api.py (hunk 1 above)
Comment (mdumandag) [reply hazelcast/serialization/api.py:2637]: OK, let me do it. I will also do the same for char and array of char

**Review by yuce — DISMISSED** [review]

**Comment by mdumandag** [comment]
@yuce can you re-approve, please?

**Review by yuce — APPROVED** [review]

**Pull request merged**